agrirouter-sdk-dotnet-standard-impl/Service/Common/HttpMessagingService.cs

```diff
@@ -62,5 +63,37 @@ public MessagingResult Send(MessagingParameters messagingParameters)
             throw new CouldNotSendHttpMessageException(httpResponseMessage.StatusCode,
                 httpResponseMessage.Content.ReadAsStringAsync().Result);
         }
+        
+        /// <summary>
+        ///     Send message to the AR using the given message parameters.
+        /// </summary>
+        /// <param name="messagingParameters">Messaging parameters.</param>
+        /// <returns>-</returns>
+        /// <exception cref="CouldNotSendHttpMessageException">Will be thrown if the message could not be send.</exception>
+        public async Task<MessagingResult> SendAsync(MessagingParameters messagingParameters)
+        {
+            var messageRequest = new MessageRequest
+            {
+                SensorAlternateId = messagingParameters.OnboardResponse.SensorAlternateId,
+                CapabilityAlternateId = messagingParameters.OnboardResponse.CapabilityAlternateId,
+                Messages = new List<Api.Dto.Messaging.Inner.Message>()
+            };
+
+            foreach (var message in messagingParameters.EncodedMessages.Select(encodedMessage =>
+                new Api.Dto.Messaging.Inner.Message
+                    {Content = encodedMessage, Timestamp = UtcDataService.NowAsUnixTimestamp()}))
+                messageRequest.Messages.Add(message);
+
+            HttpContent requestBody = new StringContent(JsonConvert.SerializeObject(messageRequest), Encoding.UTF8,
+                "application/json");
+            var httpResponseMessage = await _httpClient.PostAsync(messagingParameters.OnboardResponse.ConnectionCriteria.Measures, requestBody).ConfigureAwait(false);
+
+            if (httpResponseMessage.IsSuccessStatusCode)
+                return new MessagingResultBuilder().WithApplicationMessageId(messagingParameters.ApplicationMessageId).Build();
+
+            Log.Error($"Sending the message was not successful. HTTP response was {httpResponseMessage.StatusCode}. Please check exception for more details.");
+            throw new CouldNotSendHttpMessageException(httpResponseMessage.StatusCode,
+                httpResponseMessage.Content.ReadAsStringAsync().Result);
+        }
     }
```

Comment:
Since this is a duplicate of the already existing function I would prefer to create a method, that builds the request body.

```suggestion
        /// <summary>
        ///     Send message to the AR using the given message parameters.
        /// </summary>
        /// <param name="messagingParameters">Messaging parameters.</param>
        /// <returns>-</returns>
        /// <exception cref="CouldNotSendHttpMessageException">Will be thrown if the message could not be send.</exception>
        public async Task<MessagingResult> SendAsync(MessagingParameters messagingParameters)
        {
            var requestBody = CreateRequestBody(messagingParameters);
            var httpResponseMessage = await _httpClient.PostAsync(messagingParameters.OnboardResponse.ConnectionCriteria.Measures, requestBody).ConfigureAwait(false);

            if (httpResponseMessage.IsSuccessStatusCode)
                return new MessagingResultBuilder().WithApplicationMessageId(messagingParameters.ApplicationMessageId).Build();

            Log.Error($"Sending the message was not successful. HTTP response was {httpResponseMessage.StatusCode}. Please check exception for more details.");
            throw new CouldNotSendHttpMessageException(httpResponseMessage.StatusCode,
                httpResponseMessage.Content.ReadAsStringAsync().Result);
        }

        private static HttpContent CreateRequestBody(MessagingParameters messagingParameters)
        {
            var messageRequest = new MessageRequest
            {
                SensorAlternateId = messagingParameters.OnboardResponse.SensorAlternateId,
                CapabilityAlternateId = messagingParameters.OnboardResponse.CapabilityAlternateId,
                Messages = new List<Api.Dto.Messaging.Inner.Message>()
            };

            foreach (var message in messagingParameters.EncodedMessages.Select(encodedMessage =>
                new Api.Dto.Messaging.Inner.Message
                    {Content = encodedMessage, Timestamp = UtcDataService.NowAsUnixTimestamp()}))
                messageRequest.Messages.Add(message);

            HttpContent requestBody = new StringContent(JsonConvert.SerializeObject(messageRequest), Encoding.UTF8,
                "application/json");
            return requestBody;
        }
```

Reply:
Since the former method was not involved, the duplicate within the method should be replaced as well.

Reply:
Fixed